client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-post-setup/index.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { StepContainer, base64ImageToBlob, uploadAndSetSiteLogo } from '@automattic/onboarding';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import { FormEvent, useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSite } from '../../../../hooks/use-site';
+import SetupForm from '../link-in-bio-setup/setup-form';
+import type { Step } from '../../types';
+
+import '../link-in-bio-setup/styles.scss';
+
+const LinkInBioPostSetup: Step = function LinkInBioPostSetup( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const site = useSite();
+
+	const [ siteTitle, setComponentSiteTitle ] = useState( '' );
+	const [ tagline, setTagline ] = useState( '' );
+	const [ invalidSiteTitle, setInvalidSiteTitle ] = useState( false );
+	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
+	const [ base64Image, setBase64Image ] = useState< string | null >();
+
+	const { saveSiteSettings } = useDispatch( SITE_STORE );
+
+	useEffect( () => {
+		setComponentSiteTitle( site?.name || '' );
+		setTagline( site?.description || '' );
+	}, [ site ] );
+
+	const handleSubmit = async ( event: FormEvent ) => {
+		event.preventDefault();
+		setInvalidSiteTitle( ! siteTitle.trim().length );
+		if ( site ) {
+			await saveSiteSettings( site.ID, {
```

Comment:
`NB:` There was a delay after clicking the continue button before the next view is loaded. If the user clicks the "continue" button x number of times, x number of `sites/:siteid/settings` are dispatched to the server. Feel free to address this in a follow-up, but we'll probably want to fix it with some sort of disabled / loading state sooner rather than later

Reply:
![2022-09-16 10 32 34](https://user-images.githubusercontent.com/5414230/190698838-9af715f7-d107-46f8-8c60-1b73d50e15b6.gif)

Reply:
Yes, I actually think I should consider resolving this prior to merge.

Reply:
You can use `isResolving` selector from core/data to disable the button when things are resolving. Search for `isResolving` for examples.

Reply:
@jeyip I've added some very simple loading state at the component level to resolve this.

Reply:
Did the isResolving solution fail? It's better IMO because it will cover a new other cases (eg: if you submit another way).

Reply:
@alshakero - I was going to post a note/question about this to you. I encountered a number of different challenges with isResolving. The main one is `.isResolving()` requires that you pass in two arguments, the key for a store and the name of selector/action from that store.

But the most time consuming action we're taking within the setup form is done by the uploadAndSetSiteLogo() hook, which does not work with any store or selector. It simply makes a direct API request to update the site logo and icon.

I'm anticipating a similar issue when adding the post setup screen for Newsletters. Those require updating a site's accent color. We'll be updating an endpoint to handle that, but that action also won't involve working through a store or selector, and will rely on a direct api request.

There were also a number of secondary considerations.

You'd need to fetch and check against `isResolving` values for each item being updated. For link-in-bio, it's just two items (settings, icon/logo). For newsletter, it will be three (settings, icon/logo, accent color). But going forward, it will vary by flow, and could grow.

Also, in an earlier implementation of my current code, I found that if you end the isLoading state after the updates/requests have finished, but before the stepper submit method is invoked, you get a moment where the continue/submit button returns to active state before redirecting from the screen. This created a jumpy UI and the potential for multiple submits by the user (one of the things we were trying to avoid). Logically, I think that issue would return with isResolving, since actions would resolve momentarily before submit is invoked.

With all the above concerns in mind, I was looking for a simple solution that would allow us to deploy rest of the code we've accumulated in this PR, but based on the assumption we'd discuss and return to the loading state in a separate PR.

Reply:
Great explanation. Thank you so much! My only issue with a loading state is that it doesn't handle failure. It will be stuck disabled forever if the network drops. I think we should wrap the requests with try catch and set `isLoading` to false when an exception is thrown.

Like so 
```
const handleSubmit = async ( event: FormEvent ) => {
		event.preventDefault();
		setIsLoading( true );
		setInvalidSiteTitle( ! siteTitle.trim().length );
        try {
			  if ( site ) {
				  await saveSiteSettings( site.ID, {
					  blogname: siteTitle,
					  blogdescription: tagline,
				  } );
	  
				  if ( base64Image ) {
					  await uploadAndSetSiteLogo(
						  site.ID,
						  new File( [ base64ImageToBlob( base64Image ) ], 'site-logo.png' )
					  );
				  }
			  }
	  
			  if ( siteTitle.trim().length ) {
				  submit?.();
			  }
	  } catch (e) {
	  	setIsLoading(false);
	  }
	};
```

Reply:
@alshakero - Great suggestion. Thanks. This has been implemented. Given the potential of catching errors, I also added a bit of code to handle form errors.

Reply:
I checked it and it's great now.